### ISSUE TITLE
Pop!_OS Dist Upgrades

### ISSUE THREAD

**Issue by WatchMkr** [pop-os/upgrade#32]
Offline dist-upgrades.

* Notify user that a OS version update is available to download
* Require user action to start download (Setting > About)
* Notify user that they can start the upgrade once the package download is complete
* Download additional packages if things changed between when the packages were first downloaded and the user decided to start the upgrade.
* Upgrade UI elements in Settings > About (notification takes the user here)
* Reboot into the recovery partition and installer
* Use the progress window in the installer to show upgrade progress
* Automatically reboot once complete

Settings > About > New version available to download

![pop about settings](https://user-images.githubusercontent.com/3430283/51066611-81647e00-15c8-11e9-84c4-df9ec33e1d8f.png)

Download progress

![pop upgrade downloading](https://user-images.githubusercontent.com/3430283/51066636-9d681f80-15c8-11e9-947a-cb94479c602f.png)

Download complete

![pop upgrade download complete](https://user-images.githubusercontent.com/3430283/51066650-ab1da500-15c8-11e9-96cb-1007ba82e1e1.png)

Confirm upgrade window

![pop upgrade confirmation](https://user-images.githubusercontent.com/3430283/51066654-b1ac1c80-15c8-11e9-8ebb-e7592f473a64.png)

Running the current version

![pop upgrade current version](https://user-images.githubusercontent.com/3430283/51066657-b83a9400-15c8-11e9-9e4d-3b6fa5984ee7.png)

**Comment by brs17** [comment]
With pop-upgrade out on 18.04, 19.04, and 19.10 I think this issue can be closed. A new issue should be created to discuss adding additional features to the `About` page (Refresh install, etc).
